File: cpp/Easy/21_merge_two_sorted_lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    void insert(ListNode** head, int num) {
        ListNode* temp = new ListNode();
        temp->val = num;
        temp->next = *head;
        *head = temp;
    }
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode* merged = NULL;
        vector<int> ver;
        while(list1 != NULL) {
            ver.push_back(list1->val);
            list1 = list1->next;
        }
        while(list2 != NULL) {
            ver.push_back(list2->val);
            list2 = list2->next;
        }
        sort(ver.begin(), ver.end(), greater<int>());
        for(auto v: ver) {
            insert(&merged, v);
        }
        return merged;
    }
};

File: python/easy/21_merge_two_sorted_lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: 
Optional[ListNode]) -> Optional[ListNode]:
        temp = ListNode()
        tail = temp
        while list1 and list2:
            if list1.val < list2.val:
                tail.next = list1
                list1 = list1.next
            else:
                tail.next = list2
                list2 = list2.next
            print(tail)
            tail = tail.next
        if list1:
            tail.next = list1
        elif list2:
            tail.next = list2
        return temp.next


class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: 
Optional[ListNode]) -> Optional[ListNode]:
        merged = None
        sorted_list = list()
        while list1:
            sorted_list.append(list1.val)
            list1 = list1.next
        while list2:
            sorted_list.append(list2.val)
            list2 = list2.next
        sorted_list.sort(reverse=True)
        for i in sorted_list:
            merged = self.insert(merged, i)
        return merged

    def insert(self, merged: Optional[ListNode], num: int):
        temp = ListNode()
        temp.val = num
        temp.next = merged
        merged = temp
        return merged

